Spring version of example generated by vaadin application archetype #10.

denis-anisimov opened the pull request:
Not yet completed:
- More tests are required.
- V8 migration is almost not done: old Grid is used with old data source, FieldGroups with old fields (will be created separate issues).

Application uses Spring Boot+Spring Data+ Vaadin Spring + in memory HSQL DB.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework8-demo/10)

<!-- Reviewable:end -->

Commit:
Spring version of example generated by vaadin application archetype #10.

hesara commented:
Reviewed 14 of 57 files at r1.
Review status: 14 of 57 files reviewed at latest revision, 28 unresolved discussions.

---

_[spring-demo/pom.xml, line 22 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJp29J-flmWs5WXGYn:-KQJp29J-flmWs5WXGYo:bjbf7hy) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/pom.xml#L22)):_

> ``` XML
>     <module>spring-demo-backend</module>
>     <module>spring-demo-ui</module>
>     <module>spring-demo-production</module>
> ```

we might want to delete this module and replace it with a mention on production mode in a readme file like in https://github.com/vaadin/maven-integration/blob/master/archetypes/vaadin-archetype-application-multimodule/src/main/resources/archetype-resources/README.md (note that it seems some other parts of that readme are outdated)

---

_[spring-demo/README.md, line 40 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpCpPpDfEHLhQCS4v:-KQJpCpPpDfEHLhQCS4w:bl2atuu) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L40)):_

> ``` Markdown
>   - run "mvn vaadin:run-codeserver" in widgetset module
>   - activate Super Dev Mode in the debug window of the application
> - creating a production mode war
> ```

see the discussion in spring-demo/pom.xml on production mode; if removed, remove all references to the separate module in texts and elsewhere

---

_[spring-demo/README.md, line 61 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpXoClVMPJlfBWrBM:-KQJpXoClVMPJlfBWrBN:b-n2w1tz) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L61)):_

> ``` Markdown
> significantly.
> 
> The production module always automatically precompiles the theme for the production WAR.
> ```

see above; it seems that the readme for the multimodule archetype is also outdated

---

_[spring-demo/README.md, line 63 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpPsHa_nhtSTRJUZa:-KQJpPsHa_nhtSTRJUZb:bqrw122) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L63)):_

> ``` Markdown
> The production module always automatically precompiles the theme for the production WAR.
> 
> Using Vaadin pre-releases
> ```

(I assume this is always active in the demo project)

---

_[spring-demo/spring-demo-backend/pom.xml, line 22 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqKKiChvG4PmzrJUG:-KQJqKKiChvG4PmzrJUH:b-u0hh3v) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L22)):_

> ``` XML
>       <dependency>
>           <groupId>org.springframework.boot</groupId>
>           <artifactId>spring-boot-starter-data-jpa</artifactId>
> ```

doesn't this starter also bring in some of the other dependencies listed below? (not sure)

---

_[spring-demo/spring-demo-backend/pom.xml, line 48 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqUtpo1xKAo1f9Mdp:-KQJqUtpo1xKAo1f9Mdq:b-bklrzp) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L48)):_

> ``` XML
>       </dependency>
> 
>       <dependency>
> ```

why are these overridden and not using the starter version? perhaps add a comment if there is a good reason for this

---

_[spring-demo/spring-demo-backend/pom.xml, line 98 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqeZmxaZMVPwsOUwL:-KQJqeZmxaZMVPwsOUwM:bq5d8u8) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L98)):_

> ``` XML
>       <dependency>
>           <groupId>org.springframework.boot</groupId>
>           <artifactId>spring-boot-starter-test</artifactId>
> ```

this should probably be in the test scope

---

_[spring-demo/spring-demo-backend/pom.xml, line 103 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqr1G5Sqosps6-jaH:-KQJqr1G5Sqosps6-jaI:b-j56zud) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L103)):_

> ``` XML
>       <dependency>
>           <groupId>junit</groupId>
>           <artifactId>junit</artifactId>
> ```

test scope

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Category.java, line 29 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrQAXuaqZQtzy_J_9:-KQJrQAXuaqZQtzy_J_A:bb9eeud) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Category.java#L29)):_

> ``` Java
>     }
> 
>     public void setId(int id) {
> ```

do we need a public setId()? does Spring need it?

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Product.java, line 51 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJregPM6OH4kaHtHUb:-KQJregPM6OH4kaHtHUc:b-1olafj) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Product.java#L51)):_

> ``` Java
>     }
> 
>     public void setId(int id) {
> ```

do we need a public setId()?

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/CategoryRepository.java, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrp-flQP6BzZ0bEca:-KQJrp-flQP6BzZ0bEcb:bqkw7qm) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/CategoryRepository.java#L26)):_

> ``` Java
> 
> /**
>  * @author Vaadin Ltd
> ```

javadoc?

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/ProductRepository.java, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrr5pV05S1ZM7Jk9p:-KQJrr5pV05S1ZM7Jk9q:bqkw7qm) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/ProductRepository.java#L26)):_

> ``` Java
> 
> /**
>  * @author Vaadin Ltd
> ```

javadoc?

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 41 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJv71m53rqlsNAYob6:-KQJv71m53rqlsNAYob7:brmlzyb) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L41)):_

> ``` Java
>         Product save = insertProduct("foo");
> 
>         Assert.assertNotEquals(-1, save);
> ```

how could the Product equal -1 ?

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 72 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJvcnOZRrIIzHu79h2:-KQJvcnOZRrIIzHu79h3:b583vtc) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L72)):_

> ``` Java
>         long count = categoryRepository.count();
>         Category created = categoryRepository.save(category);
>         Assert.assertNotEquals(-1, created);
> ```

never -1, so pointless assert

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 105 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJvPmSH0FfA7vekIWM:-KQJvPmSH0FfA7vekIWN:b-exfiq) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L105)):_

> ``` Java
>     private Product insertProduct(String name) {
>         Product product = new Product();
>         product.setProductName("foo");
> ```

not using the name parameter?

---

_[spring-demo/spring-demo-ui/pom.xml, line 73 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzJaqNukR6UUjZZZC:-KQJzJaqNukR6UUjZZZD:b-sc027u) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L73)):_

> ``` XML
>           <groupId>com.vaadin</groupId>
>           <artifactId>vaadin-compatibility-shared</artifactId>
>           <version>8.0-SNAPSHOT</version>
> ```

these versions come from the BOM, so remove from here

---

_[spring-demo/spring-demo-ui/pom.xml, line 100 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzr33EjmYawc2-ZlP:-KQJzr33EjmYawc2-ZlQ:b-9r73bi) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L100)):_

> ``` XML
>   </dependencies>
> 
>   <dependencyManagement>
> ```

perhaps this should be in the parent POM as also used by the widgetset module

---

_[spring-demo/spring-demo-ui/pom.xml, line 123 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzxPMzcmdy-b9i9mO:-KQJzxPMzcmdy-b9i9mP:b-bks2a4) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L123)):_

> ``` XML
>                   <attachClasses>true</attachClasses>
>                   <!-- Exclude some unnecessary files generated by the GWT compiler. -->
>                   <packagingExcludes>WEB-INF/classes/VAADIN/gwt-unitCache/**,
> ```

not sure if all the excludes are needed with the defaults of the 7.7.0 plugin

---

_[spring-demo/spring-demo-ui/pom.xml, line 130 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJytuslh0C9gsKQfcx:-KQJytuslh0C9gsKQfcy:b-8m2inl) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L130)):_

> ``` XML
>               <groupId>com.vaadin</groupId>
>               <artifactId>vaadin-maven-plugin</artifactId>
>               <version>7.6.8</version>
> ```

use pluginManagement for version 7.7.0 in parent POM, or at least a property in the parent called vaadin.maven.version (standard naming, a lot of test scripts etc. assume that all projects accept vaadin.version and vaadin.plugin.version parameters)

---

_[spring-demo/spring-demo-ui/pom.xml, line 131 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJz1ilpMXgFGR7vf6g:-KQJz1ilpMXgFGR7vf6h:bkh2fxi) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L131)):_

> ``` XML
>               <artifactId>vaadin-maven-plugin</artifactId>
>               <version>7.6.8</version>
>               <configuration>
> ```

remove the configuration section - probably nothing needed when using 7.7.0 plugin just for updating/compiling the theme

---

_[spring-demo/spring-demo-ui/pom.xml, line 157 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQK-MR3oxvYMsD7RorE:-KQK-MR3oxvYMsD7RorF:btu9bc9) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L157)):_

> ``` XML
>                   <!--  This is required to be able to start Tomcat on Travis 
>                          and be able to manage its instance (detect that it's started)
>                          The other opition is to use Untdertow instead of Tomcat with 
> ```

Undertow

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 16 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwBzL9q-5ptsGVI-O:-KQJwBzL9q-5ptsGVI-P:b-o6u1jk) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L16)):_

> ``` XML
> 
>   <dependencies>
>       <!-- Versions for these are configured in the parent POM -->
> ```

perhaps parent should have the BOM import that is now in the ui module; then remove version numbers here; also see the comments in the UI module POM

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 46 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwNKxN6Vb4ZGsdxOI:-KQJwNKxN6Vb4ZGsdxOJ:b-ogobar) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L46)):_

> ``` XML
>               <groupId>com.vaadin</groupId>
>               <artifactId>vaadin-maven-plugin</artifactId>
>               <version>7.6.8</version>
> ```

update this to 7.7.0

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 48 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwXpj992licszjr8T:-KQJwXpj992licszjr8U:b-wisos1) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L48)):_

> ``` XML
>               <version>7.6.8</version>
>               <configuration>
>                   <extraJvmArgs>-Xmx512M -Xss1024k</extraJvmArgs>
> ```

when using 7.7.0 version of the maven plug-in, should be able to remove most of the parameters as it uses more sensible defaults

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 87 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwmgNURApQIQlbcYp:-KQJwmgNURApQIQlbcYq:b-92j9qt) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L87)):_

> ``` XML
>                       </manifestEntries>
>                   </archive>
>                   <!-- Exclude some unnecessary files generated by the GWT compiler. -->
> ```

TBC if these still need to be excluded after the plugin changes mentioned above

---

_[spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/client/samples/ResetButtonForTextFieldConnector.java, line 34 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJxMwotkUdUmsoZrxt:-KQJxMwotkUdUmsoZrxu:b-hbku5z) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/client/samples/ResetButtonForTextFieldConnector.java#L34)):_

> ``` Java
>     @Override
>     protected void extend(ServerConnector serverConnector) {
>         serverConnector.addStateChangeHandler(
> ```

(probably could make StateChangeHandler a functional interface and use a lambda here - noting here to make a reminder of going through those in the framework)

---

_[spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/samples/AttributeExtensionState.java, line 12 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJyWwa6VPcPQc0k7mI:-KQJyWwa6VPcPQc0k7mJ:bgfvljk) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/samples/AttributeExtensionState.java#L12)):_

> ``` Java
>  */
> public class AttributeExtensionState extends JavaScriptExtensionState {
>     public HashMap<String, String> attributes = new HashMap<String, String>();
> ```

(could use diamond)

---

_[spring-demo/spring-demo-widgetset/src/main/resources/com/vaadin/framework8/samples/attribute_extension_connector.js, line 12 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJyK_HfMrR6mn1F4Lc:-KQJyK_HfMrR6mn1F4Ld:b-1gzw5q) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/src/main/resources/com/vaadin/framework8/samples/attribute_extension_connector.js#L12)):_

> ``` JavaScript
>                                 element.setAttribute(attr, attributes[attr]);
>                         } catch (e) {
>                                 // IE8 does not support type='number' - just ignore it
> ```

(IE8 no longer supported, but can leave the try here in case of other problems)

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10)_

<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Review status: 14 of 57 files reviewed at latest revision, 27 unresolved discussions.

---

_[spring-demo/pom.xml, line 22 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJp29J-flmWs5WXGYn:-KQK0kBYSb-6M45-xPcp:b34xd06) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/pom.xml#L22)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> we might want to delete this module and replace it with a mention on production mode in a readme file like in https://github.com/vaadin/maven-integration/blob/master/archetypes/vaadin-archetype-application-multimodule/src/main/resources/archetype-resources/README.md (note that it seems some other parts of that readme are outdated)
> </details>
> That's weird. I've already removed it. 
> Something is wrong here. I definitely remember I did it.
> Don't know why it's still here....

---

_[spring-demo/README.md, line 40 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpCpPpDfEHLhQCS4v:-KQK2Unxt-Rsc2Wif621:b-hbdv7p) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L40)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> see the discussion in spring-demo/pom.xml on production mode; if removed, remove all references to the separate module in texts and elsewhere
> </details>
> Done.
> (I hope now)

---

_[spring-demo/README.md, line 61 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpXoClVMPJlfBWrBM:-KQK2ZiT9OTR3MSC67sv:bwby4dt) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L61)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> see above; it seems that the readme for the multimodule archetype is also outdated
> </details>
> Done.
> (I hope)

---

_[spring-demo/README.md, line 63 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJpPsHa_nhtSTRJUZa:-KQK2fX1eNpmztODLIv1:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/README.md#L63)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> (I assume this is always active in the demo project)
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/pom.xml, line 22 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqKKiChvG4PmzrJUG:-KQK2lNEc1IRYUWZHP3K:bhmuadv) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L22)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> doesn't this starter also bring in some of the other dependencies listed below? (not sure)
> </details>
> Sure, it does. And their versions are outdated or have incompatible dependencies. That's why they are listed below with more recent versions.

---

_[spring-demo/spring-demo-backend/pom.xml, line 48 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqUtpo1xKAo1f9Mdp:-KQK2xL69mbX-egOTkex:bbscrkz) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L48)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> why are these overridden and not using the starter version? perhaps add a comment if there is a good reason for this
> </details>
> As I said above: the versions are outdated or incompatible with hibernate dependencies.
> I spent a lot of time to assemble the latest Spring version with appropriate Hibernate version together.

Done.

---

_[spring-demo/spring-demo-backend/pom.xml, line 98 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqeZmxaZMVPwsOUwL:-KQK4nWmDCC2NsF-t8OR:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L98)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> this should probably be in the test scope
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/pom.xml, line 103 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqr1G5Sqosps6-jaH:-KQK53-mU2hGUWUoaZV5:b-sy30li) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L103)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> test scope
> </details>
> Its scope is defined in dependencyManagement in the root POM.
> Isn't it enough ?

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Category.java, line 29 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrQAXuaqZQtzy_J_9:-KQK5DMFWPE4FTPDH9ZX:bjulk9q) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Category.java#L29)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> do we need a public setId()? does Spring need it?
> </details>
> No this is as-is copy form the archetype application.

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Product.java, line 51 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJregPM6OH4kaHtHUb:-KQKCNfe19KrEVHXL0Fe:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/data/Product.java#L51)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> do we need a public setId()?
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/CategoryRepository.java, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrp-flQP6BzZ0bEca:-KQKD9giu23F9qXUzrTe:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/CategoryRepository.java#L26)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> javadoc?
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/ProductRepository.java, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJrr5pV05S1ZM7Jk9p:-KQKDAGsFVgnbscHW8KR:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/main/java/com/vaadin/framework8/samples/backend/repository/ProductRepository.java#L26)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> javadoc?
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 41 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJv71m53rqlsNAYob6:-KQKDWOIeddnJIYJvhzi:b6zrrkh) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L41)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> how could the Product equal -1 ?
> </details>
> Good catch.

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 72 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJvcnOZRrIIzHu79h2:-KQKDe3kpOQCr8FYMWNI:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L72)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> never -1, so pointless assert
> </details>
> Done.

---

_[spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java, line 105 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJvPmSH0FfA7vekIWM:-KQKDqUqwZ4wN8TV6eFD:bqkakev) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/src/test/java/com/vaadin/framework8/samples/backend/DataServiceTest.java#L105)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> not using the name parameter?
> </details>
> Not corrected refactoring.

---

_[spring-demo/spring-demo-ui/pom.xml, line 73 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzJaqNukR6UUjZZZC:-KQKE8GyKfPgEpxl_4ij:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L73)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> these versions come from the BOM, so remove from here
> </details>
> Done.

---

_[spring-demo/spring-demo-ui/pom.xml, line 100 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzr33EjmYawc2-ZlP:-KQKEGU0M5v8IoP-kW_2:b-th4zyz) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L100)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> perhaps this should be in the parent POM as also used by the widgetset module
> </details>
> But it's not needed in the backend module.

---

_[spring-demo/spring-demo-ui/pom.xml, line 123 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzxPMzcmdy-b9i9mO:-KQKELfV0y9d2COxprEC:bquz8z5) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L123)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> not sure if all the excludes are needed with the defaults of the 7.7.0 plugin
> </details>
> This is as-is generated POM from archetype (except added sections for Spring).  
> I can remove whatever you say.

---

_[spring-demo/spring-demo-ui/pom.xml, line 130 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJytuslh0C9gsKQfcx:-KQKF0Jq9JG60mBKQa_W:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L130)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> use pluginManagement for version 7.7.0 in parent POM, or at least a property in the parent called vaadin.maven.version (standard naming, a lot of test scripts etc. assume that all projects accept vaadin.version and vaadin.plugin.version parameters)
> </details>
> Done.

---

_[spring-demo/spring-demo-ui/pom.xml, line 131 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJz1ilpMXgFGR7vf6g:-KQKFLFETd8yaexMKcSn:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L131)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> remove the configuration section - probably nothing needed when using 7.7.0 plugin just for updating/compiling the theme
> </details>
> Done.

---

_[spring-demo/spring-demo-ui/pom.xml, line 157 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQK-MR3oxvYMsD7RorE:-KQKFWxOR68Fzkw3vNuZ:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L157)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> Undertow
> </details>
> Done.

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 16 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwBzL9q-5ptsGVI-O:-KQKJnmzKdAr7R2zMFdY:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L16)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> perhaps parent should have the BOM import that is now in the ui module; then remove version numbers here; also see the comments in the UI module POM
> </details>
> Done.

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 46 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwNKxN6Vb4ZGsdxOI:-KQKJrs8EbhLVwkLGNWr:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L46)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> update this to 7.7.0
> </details>
> Done.

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 48 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwXpj992licszjr8T:-KQKJyrvph63Te-9o9sl:b3m70ao) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L48)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> when using 7.7.0 version of the maven plug-in, should be able to remove most of the parameters as it uses more sensible defaults
> </details>
> Just let me know which parameters may be removed.
> I can't extract this information from there https://vaadin.com/download/release/7.7/7.7.0/release-notes.html#enhancements (if it's possible at all).

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 87 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwmgNURApQIQlbcYp:-KQKKofHNfNWqXn3GkNO:bsobot3) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L87)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> TBC if these still need to be excluded after the plugin changes mentioned above
> </details>
> I don't know.

---

_[spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/client/samples/ResetButtonForTextFieldConnector.java, line 34 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJxMwotkUdUmsoZrxt:-KQKNRFCdjsOFKXTEg0Y:bt0d8k0) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/client/samples/ResetButtonForTextFieldConnector.java#L34)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> (probably could make StateChangeHandler a functional interface and use a lambda here - noting here to make a reminder of going through those in the framework)
> </details>
> I've corrected it to use J8 notations. Regardless of  FW changes.

---

_[spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/samples/AttributeExtensionState.java, line 12 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJyWwa6VPcPQc0k7mI:-KQKNg1EurVH8MCgtWX2:b-896fix) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/src/main/java/com/vaadin/framework8/samples/AttributeExtensionState.java#L12)):_
<details><summary>

<i>Previously, hesara (Henri Sara) wrote…</i></summary>



> (could use diamond)
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10)_

<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Review status: 12 of 55 files reviewed at latest revision, 24 unresolved discussions.

---

_[spring-demo/spring-demo-ui/pom.xml, line 73 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzJaqNukR6UUjZZZC:-KQKVapIUpvFgeNY3ZWu:b-gvj1fe) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L73)):_
<details><summary>

<i>Previously, denis-anisimov wrote…</i></summary>



> Done.
> </details>
> Not done and I'm not going to that, sorry.
> Importing BOM in the parent POM breaks backend module because of some JARS incompatibility and I'm too tired to investigate this conflicts one more time.

---

_[spring-demo/spring-demo-widgetset/pom.xml, line 16 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJwBzL9q-5ptsGVI-O:-KQKUgWOVakMk1-bpNgO:b-gvj1fe) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-widgetset/pom.xml#L16)):_
<details><summary>

<i>Previously, denis-anisimov wrote…</i></summary>



> Done.
> </details>
> Not done and I'm not going to that, sorry.
> Importing BOM in the parent POM breaks backend module because of some JARS incompatibility and I'm too tired to investigate this conflicts one more time.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10)_

<!-- Sent from Reviewable.io -->

Artur- commented:
Reviewed 6 of 57 files at r1, 2 of 15 files at r2.
Review status: 6 of 43 files reviewed at latest revision, 16 unresolved discussions.

---

_[spring-demo/spring-demo-backend/pom.xml, line 103 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJqr1G5Sqosps6-jaH:-KQ_HE-VxZN57zhQaHvJ:b-86dvba) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-backend/pom.xml#L103)):_
<details><summary>

<i>Previously, denis-anisimov wrote…</i></summary>



> Its scope is defined in dependencyManagement in the root POM.
> Isn't it enough ?
> </details>
> It is

---

_[spring-demo/spring-demo-ui/pom.xml, line 73 [r1]](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-KQJzJaqNukR6UUjZZZC:-KQ_ug88HOyFEpo7CX5n:bzewk51) ([raw file](https://github.com/vaadin/framework8-demo/blob/a1c799f2083b31096be612b9874795a9171a60fa/spring-demo/spring-demo-ui/pom.xml#L73)):_

> Importing BOM in the parent POM breaks backend module because of some JARS incompatibility
> Imported the BOM here

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10)_

<!-- Sent from Reviewable.io -->

hesara commented:
Can make further changes later if needed - ok for now

---

Review status: 6 of 43 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-:-KQtyHEaQg2ChnhDr8HA:b38yps0)_

<!-- Sent from Reviewable.io -->

hesara commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 6 of 43 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10#-:-KQtyZ3b6vd6jGIzFiPz:bnfp4nl)_

<!-- Sent from Reviewable.io -->

hesara committed:
Merge master into spring-demo

Artur- commented:
Reviewed 26 of 57 files at r1, 5 of 15 files at r2, 7 of 21 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework8-demo/10)_

<!-- Sent from Reviewable.io -->
